src/toil/job.py

```diff
@@ -1419,6 +1419,23 @@ def setupJobAfterFailure(
                 self,
                 self.jobStoreID,
             )
+        elif (
+            exit_reason
+            in (
+                BatchJobExitReason.MEMLIMIT,
+                BatchJobExitReason.PKILL,
```

Comment:
```suggestion
                BatchJobExitReason.PARTITION,
```
as exit code 9 is being treated as PARTITION issue by toil devs.

Reply:
Noted